src/main/java/user_code_resolver/UserCodeResolver.java

```diff
@@ -0,0 +1,222 @@
+                }
+            } catch (IOException e) {
+                System.out.println("An error occurred while writing the file: " + e.getMessage());
+            }
+        }
+    }
+
+
+    private Project updateNamesOfImplFiles(Project project, File folder) throws IOException {
+        List<String> projectStructureList = new ArrayList<>();
+        List<File> fileList = readStructureFromFolderAsList(folder);
+        for (UserFiles userList : project.getFiles()) {
+            projectStructureList.add(userList.getFilename());
+        }
+        Diff diff = compareStructure(projectStructureList, convertNewProjectStructureToList(fileList));
+        for (Change change : diff.getChanges()) {
+            if (change instanceof ListChange listChange) {
+                for (ContainerElementChange element : listChange.getChanges()) {
+                    if (element instanceof ElementValueChange elementValueChange) {
+                        String oldPath = elementValueChange.getLeftValue().toString();
+                        String newPath = elementValueChange.getRightValue().toString();
+                        String[] oldPathArray = oldPath.split("\\\\");
+                        String[] newPathArray = newPath.split("\\\\");
+                        int fileTypeStart = oldPathArray[oldPathArray.length - 1].indexOf(".");
+                        String oldConstructorName = oldPathArray[oldPathArray.length - 1].substring(0, fileTypeStart);
+                        String newConstructorName = newPathArray[oldPathArray.length - 1].substring(0, fileTypeStart);
+                        String newPackage = oldPathArray[oldPathArray.length - 2];
+                        for (UserFiles file : project.getFiles()) {
+                            if (file.getFilename() == elementValueChange.getLeftValue()) {
+                                file.setFilename(newPath);
+                                for (int i = 0; i < file.getContent().size(); i++) {
+                                    String line = file.getContent().get(i);
+                                    if (line == null) continue;
+                                    if (line.contains("package")) {
+                                        for (int arrayIndex = oldPathArray.length - 3; arrayIndex >= 0; arrayIndex--) {
+                                            if (oldPathArray[arrayIndex].equals("java")) {
+                                                break;
+                                            } else {
+                                                newPackage = oldPathArray[arrayIndex] + "." + newPackage;
+                                            }
+                                        }
+                                        line = "package " + newPackage + ";";
+                                        file.getContent().set(i, line);
+                                    } else if (line.contains("class ") || line.contains(oldConstructorName)) {
```

Comment:
Hier statt `else if` lieber ein neues `if`. Es wird ja nicht vorkommen dass in einer Zeile class + name und package zusammen stehen oder? `else ifs` erschweren mMn den Lesefluss